stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -148,6 +150,28 @@ internal class ChannelLogic(
             }
     }
 
+    override suspend fun onHideChannelRequest(channelType: String, channelId: String, clearHistory: Boolean) = setHidden(true)
+
+    override suspend fun onHideChannelResult(
+        result: Result<Unit>,
+        channelType: String,
+        channelId: String,
+        clearHistory: Boolean,
+    ) {
+        if (result.isSuccess) {
+            val cid = Pair(channelType, channelId).toCid()
+            if (clearHistory) {
+                val now = Date()
+                mutableState.hideMessagesBefore = now
+                removeMessagesBefore(now)
+                chatDomainImpl.repos.deleteChannelMessagesBefore(cid, now)
+                chatDomainImpl.repos.setHiddenForChannel(cid, true, now)
+            } else {
+                chatDomainImpl.repos.setHiddenForChannel(cid, true)
+            }
+        }
```

Comment:
Shouldn't we also call `setHidden(false)` if request failed? 🤔 Maybe we should only `setHidden(true)` when we get the result from the API?

Reply:
In old logic, we don't do it. I am not sure why we didn't do it earlier?